internal/pkg/bootstrap/handlers/database/database.go

```diff
@@ -104,7 +105,14 @@ func (d Database) BootstrapHandler(
 	var dbClient dbInterfaces.DBClient
 	for startupTimer.HasNotElapsed() {
 		var err error
-		dbClient, err = d.newDBClient(loggingClient)
+		credentialProvider := container.CredentialsFrom(dic.Get)
+		dbCredentials, err := credentialProvider.GetDatabaseCredentials(d.database.GetDatabaseInfo()["Primary"])
+		if err != nil {
+			loggingClient.Error(fmt.Sprintf("unable to get database credentials: %s", err.Error()))
+			return false
```

Comment:
Is this truly a fatal error that should cause the bootstrapping process to be aborted and the service to be stopped?  Will a retry potentially succeed?

Reply:
I do not think we have enough information to determine if the type of error can be resolved with a retry, so I think we can still retry since we are protected by the time limit. Overall, I think it would be best if the errors had some more information where we make a better decision.

Here are the errors that will eventually bubble up: https://github.com/edgexfoundry/go-mod-secrets/blob/master/pkg/types.go

Reply:
Also, I separated the retrieval of credentials into its own retry-loop, so that if we fail to connect to the database we do not throw away the credentials that we already obtained.